_posts/2025-03-20-phased-strimzi-upgrade-example.md

```diff
@@ -0,0 +1,637 @@
+  clusterrolebinding strimzi-cluster-operator-namespaced \
+  --clusterrole=strimzi-cluster-operator-namespaced \
+  --serviceaccount strimzi:strimzi-cluster-operator \
+  > install-files/cluster-operator-0.43.0/strimzi-cluster-operator-namespaced.yaml
+```
+
+We can then add these additional resources to the `resources` list in the respective `kustomization.yaml` files and they will be modified by the same `nameSuffix` and `patch-role-reference.yaml` patches as the other RBAC resources.
+
+#### Operator Set
+
+After all these steps we now have a pattern for installing multiple versions of Strimzi at the same time without them interfering with one another, we can call this group of Strimzi versions the "Operator Set".
+
+We have a folder for each Strimzi version's install resources (without the CRDs) and the folder containing the CRDs from the **most recent** Strimzi version.
+
+`kustomization.yaml` files define the patches to the base installation files needed to individualise each version's installation files and a top level `kustomization.yaml` imports each version and the CRDs:
+
+```shell
+install-files
+├── cluster-operator-0.43.0
+│  ├── patches
+│  │  ├── patch-deployment.yaml
+│  │  └── patch-role-references.yaml
+│  ├── 010-ServiceAccount-strimzi-cluster-operator.yaml
+│  ├── ...
+│  ├── 060-Deployment-strimzi-cluster-operator.yaml
+│  ├── kustomization.yaml
+├── cluster-operator-0.44.0
+│  ├── patches
+│  │  ├── patch-deployment.yaml
+│  │  └── patch-role-references.yaml
+│  ├── 010-ServiceAccount-strimzi-cluster-operator.yaml
+│  ├── ...
+│  ├── 060-Deployment-strimzi-cluster-operator.yaml
+│  ├── kustomization.yaml
+├── cluster-operator-0.45.0
+│  ├── patches
+│  │  ├── patch-deployment.yaml
+│  │  └── patch-role-references.yaml
+│  ├── 010-ServiceAccount-strimzi-cluster-operator.yaml
+│  ├── ...
+│  ├── 060-Deployment-strimzi-cluster-operator.yaml
+│  ├── kustomization.yaml
+├── crds
+│  ├── 04A-Crd-kafkanodepool.yaml
+│  ├── 040-Crd-kafka.yaml
+│  ├── 041-Crd-kafkaconnect.yaml
+│  ├── 042-Crd-strimzipodset.yaml
+│  ├── 043-Crd-kafkatopic.yaml
+│  ├── 044-Crd-kafkauser.yaml
+│  ├── 045-Crd-kafkamirrormaker.yaml
+│  ├── 046-Crd-kafkabridge.yaml
+│  ├── 047-Crd-kafkaconnector.yaml
+│  ├── 048-Crd-kafkamirrormaker2.yaml
+│  ├── 049-Crd-kafkarebalance.yaml
+│  └── kustomization.yaml
+└── kustomization.yaml
+```
+
+To install the entire Operator Set you simply run:
+
+```shell
+kubectl -n strimzi apply -k install-files
+```
+
+Note that the namespace you install the Operator set into needs to match the one you have used in the `patch-role-references.yaml`.
+
+A full example of the Operator Set above can be seen in the [strimzi-phased-upgrade](https://github.com/tomncooper/strimzi-phased-upgrade) repository.
+That repository also contains scripting to automate the addition of new Strimzi versions and templates for the generation of patch and kustomization files. 
+
+### The Phased Upgrade Process
```

Comment:
As I said above, I think you should maybe operate with just two versions as three might become too much.